src/main/java/com/google/cloud/tools/managedcloudsdk/internal/install/Installer.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.managedcloudsdk.internal.install;
+
+import java.nio.file.Path;
+import java.util.concurrent.Callable;
+
+/**
+ * Installer for running install scripts in a Cloud SDK download. It is a callable we use in chained
+ * asynchronous calls.
+ */
+public interface Installer extends Callable<Path> {
```

Comment:
What does this give us over just implementing Callable<Path> directly?

Reply:
It is only really allowing for better documentation, by making it an `Installer` type, we can document the requirements of the call method. It is internal though, so maybe we don't need to at all. Once these 3 CLs (download/extract/install) are done, perhaps the glue class that bring them all together will make the factories and interfaces unnecessary.